examples/mem_example/src/main.cpp

```diff
@@ -22,47 +22,103 @@
 #include <stdio.h>
 #include <unistd.h>
 #include <thread>
+#include <iostream>
 
-#include <klepsydra/core/event_emitter_middleware_provider.h>
+#include <klepsydra/high_performance/event_loop_middleware_provider.h>
 #include <klepsydra/mem_vision_ocv/image_multiplexer_vision_ocv_provider.h>
+#include <klepsydra/vision_ocv/file_image_stream_service.h>
 
 #include <klepsydra/mem_core/mem_env.h>
 
 #include <image_processing/image_processor_service.h>
 
 #include "server_user_input.h"
 
-#include "simple_write_service.h"
-
 #include "config.h"
 
+#include "spdlog/spdlog.h"
+
 int main(int argc, char* argv[]) {
     std::string containerName = argc > 1 ? argv[1] : "sample1";
     std::string folderName = argc > 4 ? argv[4] : TEST_DATA;
     kpsr::mem::MemEnv environment;
+    //spdlog::set_level(spdlog::level::debug);
+    const int nImageProcessingServices = 5;
+    int ctrImageProcessingServices[nImageProcessingServices] = {0, 0, 0, 0, 0};
 
     kpsr::vision_ocv::high_performance::ImageMultiplexerVisionMiddlewareProvider<128> provider(nullptr, "image_processing_feed",
                                                                                               1024, 640, 16, "body");
 
-    kpsr::EventEmitterMiddlewareProvider<kpsr_tutorial::ResultData> eventEmitter(nullptr,
-                                                                        "image_processing_results",
-                                                                        0, nullptr, nullptr);
+    kpsr::high_performance::EventLoopMiddlewareProvider<16> eventloop(nullptr);
+    
+    eventloop.start();
+
+    kpsr::vision_ocv::FileImageStreamingService fileImageStreamingService(&environment,
+                                                                          provider.underlying->getPublisher(),
+                                                                          folderName,
+                                                                          true,
+                                                                          480,
+                                                                          320,
+                                                                          "body",
+                                                                          "FileImageStreamingService");
+
+    kpsr_tutorial::ImageProcessingService imageProcessingService_1(&environment, 50, provider.underlying->getSubscriber(),
+                                                                   eventloop.getPublisher<kpsr_tutorial::ResultData>("image_processing_results_1", 0, nullptr, nullptr));
+    eventloop.getSubscriber<kpsr_tutorial::ResultData>("image_processing_results_1")->registerListener("image_processing_results_1", [&ctrImageProcessingServices](const kpsr_tutorial::ResultData & message) {
```

Comment:
Rather than passing "image_processing_results_1" (and the other strings) here multiple times, make a const std::string that stores the value, and pass that string instead. This will ensure that you guarantee that the string is same here and also when you remove the listener.

Reply:
Done